### PR TITLE
Use Jekyll SEO Tag and Jekyll Feed

### PR DESCRIPTION
**Pull request by benbalter**
This pull request changes the head template to use [Jekyll SEO Tag](https://github.com/jekyll/jekyll-seo-tag) and [Jekyll Feed](https://github.com/jekyll/jekyll-feed), rather than hard coding the SEO meta and feed template into the site. The idea for both being (A) pulling code out of the site where possible, (B) relying on battle-tested templates that can account for more edge cases, rather than custom logic, and (C) more open source, more better. :smile:

### Jekyll feed

There should be little or no change in behavior here, other than using a more widely tested template that can account for all sorts of edge cases, like relative URLs in posts. As an added bonus, it'll now include author information, thumbnails, and a short summary/description in addition to the full post text.

#### Before

before

#### After

after

### Jekyll SEO Tag

Again, the actual change in output should be subtle, but it lets you pull a bunch of code out of the templates and benefit from more widely tested logic/best practices, and subsequent upstream improvements. The big change is the addition of JSON-LD metadata for richer indexing, and the inclusion of author information for posts.

#### Homepage before

```html
<title>Electron</title>
<meta name='description' content='Build cross platform desktop apps with JavaScript, HTML, and CSS.'>

<meta name="google-site-verification" content="eOAFtDphTbbm4OPKva2d3Z0Z_2bBxWMGdkD0IRQ6VeA" />
<meta name='twitter:card' content='summary_large_image' />
<meta name='twitter:site' content='@electronjs' />
<meta name="twitter:creator" content="@github">
<meta name='twitter:title' content='Electron' />
<meta name='twitter:description' content='Build cross platform desktop apps with JavaScript, HTML, and CSS.' />
<meta name='twitter:image' content='http://electron.atom.io/images/opengraph.png' />
<meta property='og:title' content='Electron' />
<meta property='og:description' content='Build cross platform desktop apps with JavaScript, HTML, and CSS.' />
<meta property='og:type' content='website' />
<meta property='og:url' content='http://electron.atom.io/' />
<meta property='og:image' content='http://electron.atom.io/images/opengraph.png' />
<meta property="og:image:width" content="1200" />
<meta property="og:image:height" content="630" />
<meta property='og:site_name' content='Electron' />
<meta content='#2f3241' name='theme-color' />
<link rel='shortcut icon' href='/images/favicon.ico'/>
<link rel='stylesheet' href='/css/index.css'>
<link rel='canonical' href='http://electron.atom.io/'>
<link rel='alternate' type='application/rss+xml' title='Electron' href='http://electron.atom.io/feed.xml' />
```

#### Homepage after

```html
<!-- Begin Jekyll SEO tag v1.4.0 -->
<title>Electron - Build cross platform desktop apps with JavaScript, HTML, and CSS.</title>
<meta property="og:title" content="Electron" />
<meta name="description" content="Build cross platform desktop apps with JavaScript, HTML, and CSS." />
<meta property="og:description" content="Build cross platform desktop apps with JavaScript, HTML, and CSS." />
<link rel="canonical" href="http://electron.atom.io/" />
<meta property="og:url" content="http://electron.atom.io/" />
<meta property="og:site_name" content="Electron" />
<meta property="og:image" content="http://electron.atom.io/images/opengraph.png" />
<meta property="og:image:height" content="630" />
<meta property="og:image:width" content="1200" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2016-05-11T10:30:07-04:00" />
<link rel="next" href="http://electron.atom.io/releases/" title="Electron Releases" />
<link rel="prev" href="http://electron.atom.io/docs/guides/" title="Electron Guides" />
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:site" content="@ElectronJS" />
<meta name="google-site-verification" content="eOAFtDphTbbm4OPKva2d3Z0Z_2bBxWMGdkD0IRQ6VeA" />
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "WebSite",
  "name": "Electron",
  "headline": "Electron",
  "image": "http://electron.atom.io/images/opengraph.png",
  "datePublished": "2016-05-11T10:30:07-04:00",
  "description": "Build cross platform desktop apps with JavaScript, HTML, and CSS.",
  "logo": "http://electron.atom.io/images/opengraph.png",
  "url": "http://electron.atom.io/"
}
</script>
<!-- End Jekyll SEO tag -->
<link type="application/atom+xml" rel="alternate" href="http://electron.atom.io/feed.xml" title="Electron" />
```

#### Post before

```html
<title>Electron 1.0</title>
<meta name='description' content='For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormi...'>

<meta name="google-site-verification" content="eOAFtDphTbbm4OPKva2d3Z0Z_2bBxWMGdkD0IRQ6VeA" />
<meta name='twitter:card' content='summary_large_image' />
<meta name='twitter:site' content='@electronjs' />
<meta name="twitter:creator" content="@github">
<meta name='twitter:title' content='Electron 1.0' />
<meta name='twitter:description' content='For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormi...' />
<meta name='twitter:image' content='http://electron.atom.io/images/opengraph.png' />
<meta property='og:title' content='Electron 1.0' />
<meta property='og:description' content='For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormi...' />
<meta property='og:type' content='website' />
<meta property='og:url' content='http://electron.atom.io/blog/2016/05/11/electron-1-0' />
<meta property='og:image' content='http://electron.atom.io/images/opengraph.png' />
<meta property="og:image:width" content="1200" />
<meta property="og:image:height" content="630" />
<meta property='og:site_name' content='Electron' />
<meta content='#2f3241' name='theme-color' />
<link rel='shortcut icon' href='/images/favicon.ico'/>
<link rel='stylesheet' href='/css/index.css'>
<link rel='canonical' href='http://electron.atom.io/blog/2016/05/11/electron-1-0'>
<link rel='alternate' type='application/rss+xml' title='Electron' href='http://electron.atom.io/feed.xml' />
```

#### Post after

```html
<!-- Begin Jekyll SEO tag v1.4.0 -->
<title>Electron 1.0 - Electron</title>
<meta property="og:title" content="Electron 1.0" />
<meta name="description" content="For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormilestone for our framework and for the community that created it. The releaseof Electron 1.0 is now available from electron.atom.io." />
<meta property="og:description" content="For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormilestone for our framework and for the community that created it. The releaseof Electron 1.0 is now available from electron.atom.io." />
<link rel="canonical" href="http://electron.atom.io/blog/2016/05/11/electron-1-0" />
<meta property="og:url" content="http://electron.atom.io/blog/2016/05/11/electron-1-0" />
<meta property="og:site_name" content="Electron" />
<meta property="og:image" content="http://electron.atom.io/images/opengraph.png" />
<meta property="og:image:height" content="630" />
<meta property="og:image:width" content="1200" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2016-05-11T00:00:00-04:00" />
<link rel="prev" href="http://electron.atom.io/blog/2016/03/25/electron-37" title="What&#39;s new in Electron 0.37" />
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:site" content="@ElectronJS" />
<meta name="twitter:creator" content="@jlord" />
<meta name="google-site-verification" content="eOAFtDphTbbm4OPKva2d3Z0Z_2bBxWMGdkD0IRQ6VeA" />
<script type="application/ld+json">
  {
    "@context": "http://schema.org",
    "@type": "BlogPosting",
    "headline": "Electron 1.0",
    "image": "http://electron.atom.io/images/opengraph.png",
    "datePublished": "2016-05-11T00:00:00-04:00",
    "description": "For the last two years, Electron has helped developers build cross platformdesktop apps using HTML, CSS, and JavaScript. Now we’re excited to share a majormilestone for our framework and for the community that created it. The releaseof Electron 1.0 is now available from electron.atom.io.",
    "logo": "http://electron.atom.io/images/opengraph.png",
    "url": "http://electron.atom.io/blog/2016/05/11/electron-1-0"
  }
</script>
<!-- End Jekyll SEO tag -->
  <link type="application/atom+xml" rel="alternate" href="http://electron.atom.io/feed.xml" title="Electron" />
```

### Other changes

Beyond adding the two plugins, I ran `bundle update`, hence the updated Gemfile.lock noise.

Last, by adding `:group => :jekyll_plugins` to the Gemfile, the Pages Gem will set configuration defaults for you, so we were able to remove some default from the site config.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -6,15 +6,11 @@ twitter_username: ElectronJS
 github_username: electron
 future: true
 excerpt_separator: '---'
-highlighter: rouge
```
Comment (kevinsawicki) [inline _config.yml]: Are these changes related to the SEO/feed changes?

Comment (benbalter) [inline _config.yml]: No.

> Last, by adding :group => :jekyll_plugins to the Gemfile, the Pages Gem will set configuration defaults for you, so we were able to remove some default from the site config.

Glad to revert or submit as a separate PR.

**Comment by leereilly** [comment]
Some quick feedback (haven't pulled this down locally to test):

**1.** The `og:type` for the *homepage* should be <kbd>website</kbd> (and I think it's safe to drop `article:published_time` too).

**2.** The blog post authors in the posts are *GitHub* handles. They're not guaranteed to be the same as the author's Twitter handle. E.g. the following `twitter:creator` markup would be inaccurate as `@jlord`'s handle on Twitter is `@jllord`:

```html
<meta name="twitter:creator" content="@jlord" />
```

**3.** I don't think it makes sense to include the prev/next link tags on the static pages here. E.g. the individual guides (as they are now at least) are grouped, ordered alphabetically, and not really a paginated collection:

```html
<link rel="next" href="http://electron.atom.io/releases/" title="Electron Releases" />
<link rel="prev" href="http://electron.atom.io/docs/guides/" title="Electron Guides" />
```

**Comment by benbalter** [comment]
>  I don't think it makes sense to include the prev/next link tags on the static pages here

The behavior of these is going to change via https://github.com/jekyll/jekyll-seo-tag/pull/87.

>  The og:type for the homepage should be website (and I think it's safe to drop article:published_time too).

The issue is that the site's using a collection called `_pages` to group pages. Which [isn't how the feature's intended to be used](http://ben.balter.com/2015/02/20/jekyll-collections/), but didn't think reorganizing the site was within the scope of this pull request.

>  The blog post authors in the posts are GitHub handles. They're not guaranteed to be the same as the author's Twitter handle. 

We can add Twitter handles, if we want, the plugin supports it. Again, didn't think it was within the scope of this PR, but glad to add 'em.

**Comment by kevinsawicki** [comment]
> . Again, didn't think it was within the scope of this PR, but glad to add 'em.

It feels within scope of this PR since before this information wasn't present and now in this PR it is present but inaccurate

**Comment by kevinsawicki** [comment]
@benbalter :wave: any updates here regarding the issues @leereilly raised? Each of the three issues he raised seem worth addressing in this pull request. Thanks.

**Comment by benbalter** [comment]
>  any updates here regarding the issues @leereilly raised?

* **Next/Previous** - Will be addressed in the next release of Jekyll SEO via https://github.com/jekyll/jekyll-seo-tag/pull/87.

* **Twitter handles** - Will add them to the PR

* **Type** - Will update the PR to use pages as pages, rather than a collection called pages, unless there's any reason not to do so.

**Comment by benbalter** [comment]
Address Twitter handles, type, and date published via https://github.com/electron/electron.atom.io/pull/248/commits/059ba663acfb65a82a16767fb7e22d8f5fd9dacd and https://github.com/electron/electron.atom.io/pull/248/commits/9f66f701fc5b2c5c9df3bd234a18a2266faaf2ec.

Will get https://github.com/jekyll/jekyll-seo-tag/pull/87 on to Pages shortly, which, once it does, should address all of the above concerns.

**Comment by benbalter** [comment]
> Will get jekyll/jekyll-seo-tag#87 on to Pages shortly, which, once it does, should address all of the above concerns.

Pages now runs Jekyll SEO Tag v2.0.0, which includes https://github.com/jekyll/jekyll-seo-tag/pull/87. 

Believe this should be good to go from my point of view.

**Comment by kevinsawicki** [comment]
> Believe this should be good to go from my point of view.

@benbalter did this pull request change the `<title>` tags for each page?

When I run `script/server` locally, the main page's title is now `<title>Electron - Electron</title>`. On the current site it generates as just `<title>Electron</title>`

**Comment by benbalter** [comment]
> did this pull request change the <title> tags for each page?

Great catch. With 1157530, the homepage title is now `Electron - Build cross platform desktop apps with JavaScript, HTML, and CSS.`.

For the other pages, since many all already in the form of "Election X", we have two options:

1. Tell the plugin not to touch the titles and blindly output `<title>{{ page.title }}</tittle>` for all pages. This wouldn't help pages like http://electron.atom.io/docs/tutorial/application-distribution/, which would just have a title of "Application Distribution" (unless we updated them all to be, e.g., "Electron Application Distribution")

2. Change the page titles to just X, e.g., "Guides", "Releases", and let the plugin append " - Electron" so the titles would be, e.g., "Releases - Electron" and "Guides - Electron". I find this format makes tab titles a bit more friendly, and it'd help the docs pages, but it's really a matter of preference.

**Comment by kevinsawicki** [comment]
Awesome, looks great to now, I'll create a follow on pull request that drops the `Electron` prefix from the pages you mentioned.

Thanks so much for this @benbalter, really appreciate it 👍 💚 🚢

**Comment by benbalter** [comment]
My pleasure. Sorry it took so long, and thanks for helping to make the plugin better for users (and other GitHub sites).

One thing to note, it looks like the auto-generated excerpt is off for some of the docs. In the config, we've got `excerpt_separator: "---"`, which means it's trying to split the docs on the first HR, which doesn't always exist (thus the full body is being crammed into the description). 

The best solution, would be to remove that flag (I don't see the custom excerpt being used anywhere), which would default things to "\n\n", and allow the first paragraph to become the auto-generated excerpt when one isn't manually provided.

*But*... the docs have the title as an H1 in the doc body, meaning, the autogenerated excerpt and the title would be the same. Unless the docs are used elsewhere, glad to submit a PR to remove the H1 from the document body, and update the template to include `<h1>{{ page.title }}</h1>` if you're interested, which should allow Jekyll to properly grab the first paragraph as an excerpt for descriptions.

**Comment by kevinsawicki** [comment]
> One thing to note, it looks like the auto-generated excerpt is off for some of the docs.

Yeah, good point, thanks for mentioning that, will take a look, /cc @jlord

**Comment by zeke** [comment]
Thanks for adding this stuff, @benbalter!
